libimage/pull.go

```diff
@@ -52,6 +52,11 @@ func (r *Runtime) Pull(ctx context.Context, name string, pullPolicy config.PullP
 		options = &PullOptions{}
 	}
 
+	// Docker compat: strip off the tag iff name is tagged and digested
+	// (e.g., fedora:latest@sha256...).  In that case, the tag is stripped
+	// off and entirely ignored.  The digest is the sole source of truth.
+	name = normalizeTaggedDigestedString(name)
+
 	ref, err := alltransports.ParseImageName(name)
 	if err != nil {
 		// If the image clearly refers to a local one, we can look it up directly.
```

Comment:
… I might be missing something; how do short names like `busybox` vs explicit `docker://docker.io/busybox` pulls work? AFAICS this code parses both into a `docker://busybox` `ImageReference`, losing the distinction — and then may try to do short name lookup on explicit `docker://…` formats, which should, I think, _never_ happen — short name lookup is used only for Docker-like non-transport-qualified formats, so that the c/image-transport-qualified strings always mean the same thing in all callers.